hummingbot/market/binance/binance_market.pyx

```diff
@@ -975,7 +976,7 @@ cdef class BinanceMarket(MarketBase):
                 order_result = await self.query_api(self._binance_client.order_limit_buy,
                                                     symbol=symbol,
                                                     quantity=str(decimal_amount),
-                                                    price=price,
+                                                    price=format_float_positional(price),
```

Comment:
Please use `c_quantize_order_price()` instead. Don't hack this with a random numpy function since you may go over the maximum allowed precision and cause order failure.